src/landing/plugin.ts

```diff
@@ -32,107 +36,113 @@ const landingExtension: JupyterLabPlugin<void> = {
   autoStart: true
 };
 
+class LandingModel extends VDomModel {
+  get path(): string {
+    return this._path;
+  }
 
-function activateLanding(app: JupyterLab, services: IServiceManager, pathTracker: IPathTracker, palette: ICommandPalette): void {
-  let widget = new Widget();
-  widget.id = 'landing-jupyterlab';
-  widget.title.label = 'Launcher';
-  widget.title.closable = true;
-  widget.addClass('jp-Landing');
-
-  let dialog = document.createElement('div');
-  dialog.className = 'jp-Landing-dialog';
-  widget.node.appendChild(dialog);
-
-  let logo = document.createElement('span');
-  logo.className = 'jp-ImageJupyterLab jp-Landing-logo';
-  dialog.appendChild(logo);
-
-  let previewMessages = ['super alpha preview', 'very alpha preview', 'extremely alpha preview', 'exceedingly alpha preview', 'alpha alpha preview'];
-  let subtitle = document.createElement('span');
-  let index = Math.floor(Math.random() * previewMessages.length);
-  subtitle.textContent = previewMessages[index];
-  subtitle.className = 'jp-Landing-subtitle';
-  dialog.appendChild(subtitle);
-
-  let tour = document.createElement('span');
-  tour.className = 'jp-Landing-tour';
-  dialog.appendChild(tour);
-  tour.addEventListener('click', () => {
-    app.commands.execute('about-jupyterlab:show', void 0);
-  });
-
-  let header = document.createElement('span');
-  header.textContent = 'Start a new activity';
-  header.className = 'jp-Landing-header';
-  dialog.appendChild(header);
-
-  let body = document.createElement('div');
-  body.className = 'jp-Landing-body';
-  dialog.appendChild(body);
-
-  for (let name of ['Notebook', 'Code Console', 'Terminal', 'Text Editor']) {
-    let column = document.createElement('div');
-    body.appendChild(column);
-    column.className = 'jp-Landing-column';
-
-    let img = document.createElement('span');
-    let imgName = name.replace(' ', '');
-    img.className = `jp-Image${imgName} jp-Landing-image`;
-
-    column.appendChild(img);
-
-    let text = document.createElement('span');
-    text.textContent = name;
-    text.className = 'jp-Landing-text';
-    column.appendChild(text);
+  set path(value: string) {
+    this._path = value;
+    this.stateChanged.emit(void 0);
   }
 
-  let img = body.getElementsByClassName('jp-ImageNotebook')[0];
-  img.addEventListener('click', () => {
-    app.commands.execute('file-operations:new-notebook', void 0);
-  });
+  get data(): VNode[] {
```

Comment:
In general, the model should only have state related things and shouldn't know about the virtual DOM in any way. I would move the code that builds vdom elements using the actual model data to the VDomWidget subclass.